app/controllers/trips_controller.rb

```diff
@@ -0,0 +1,91 @@
+class TripsController < ApplicationController
+  def show
+    trip_id = params[:id].to_i
+    @trip = Trip.find_by(id: trip_id)
+
+    if @trip.nil?
+      render file: "#{Rails.root}/public/404.html", status: :not_found
+      return
+    end
+
+  end
+
+
+  def create
+    if params[:passenger_id].nil?
+      @trip = Trip.new(trip_params)
```

Comment:
It is not clear to me how you would ever receive trip params as there is not form the create a trip.